lib/rules/init-declarations.js

```diff
@@ -105,15 +107,39 @@ module.exports = {
 		const mode = context.options[0] || MODE_ALWAYS;
 		const params = context.options[1] || {};
 
+		// Track whether we're inside a declared namespace
+		let insideDeclaredNamespace = false;
+
 		//--------------------------------------------------------------------------
 		// Public API
 		//--------------------------------------------------------------------------
 
 		return {
+			TSModuleDeclaration(node) {
+				if (node.declare) {
+					insideDeclaredNamespace = true;
+				}
+			},
+
+			"TSModuleDeclaration:exit"(node) {
+				if (node.declare) {
+					insideDeclaredNamespace = false;
+				}
+			},
+
 			"VariableDeclaration:exit"(node) {
 				const kind = node.kind,
 					declarations = node.declarations;
 
+				if (mode === MODE_ALWAYS) {
+					if (node.declare) {
+						return;
+					}
+					if (insideDeclaredNamespace) {
+						return;
+					}
+				}
```

Comment:
`declare`d variables circumvent the rule anyway, so I don't think we need to check the mode here. Tests don't fail if we remove the `if`:

```suggestion
				if (node.declare || insideDeclaredNamespace) {
					return;
				}
```

Reply:
Done.